2333/2/libs/top_words_analyzer.rb

```diff
@@ -0,0 +1,31 @@
+require_relative 'handler'
+require 'terminal-table'
+require_relative './analyzer_printer.rb'
+require_relative './names_searcher.rb'
+
+# VersusBattle text analyzer
+# :reek:ControlParameter
+class TopWordsAnalyzer
+  def initialize(rappers, selected_name, take_value)
+    @rappers = rappers
+    @name = selected_name
+    @take_value = take_value || 30
+  end
+
+  def top_words
+    exit unless names_include_name?
```

Comment:
Look your `TopWordsAnalyzer` already knows about `AnalyzerPrinter`. I think it's better not to link one another class with `AnalyzerPrinter` as you did in `NameSearcher`.

Let's implement all the logic here, in this method. If there's no raper with the provided name -- we output all raper names if the name exists in our list -- we print the list of top words. So `NameSearcher` knows nothing about output.